src/app/ApplicationTemplate.Shared/Configuration/NavigationCoreConfiguration.cs

```diff
@@ -37,10 +38,22 @@ public static IServiceCollection AddNavigationCore(this IServiceCollection servi
 					var manager = new BackButtonManager();
 
 					var sectionsNavigator = s.GetRequiredService<ISectionsNavigator>();
+
 					manager.AddHandler(new BackButtonHandler(
 						name: "DefaultSectionsNavigatorHandler",
 						canHandle: () => sectionsNavigator.CanNavigateBackOrCloseModal(),
-						handle: async ct => await sectionsNavigator.NavigateBackOrCloseModal(ct)
+						handle: async ct =>
+						{
+							if (sectionsNavigator.State?.HasActiveViewModel() ?? false && sectionsNavigator.GetActiveViewModel() is ViewModel)
+							{
+								var viewModel = sectionsNavigator.GetActiveViewModel() as ViewModel;
+								await viewModel.NavigateBack.Execute(ct);
+							}
+							else
```

Comment:
This used to be so that the NavigateBack field could be set as virtual in the ViewModel and that you could override the NavigateBack to get a new BackButton behavior, it was useful at one point, but is not necessary and will not be part of the final PR.